modules/http/http-grammar.ym

```diff
@@ -172,10 +173,16 @@ http_tls_option
     | KW_CERT_FILE  '(' path_check ')'        { http_dd_set_cert_file(last_driver, $3); free($3); }
     | KW_KEY_FILE   '(' path_secret ')'       { http_dd_set_key_file(last_driver, $3); free($3); }
     | KW_CIPHER_SUITE '(' string ')'          { http_dd_set_cipher_suite(last_driver, $3); free($3); }
-    | KW_USE_SYSTEM_CERT_STORE '(' yesno ')'  { http_dd_set_use_system_cert_store(last_driver, $3); }
     | KW_SSL_VERSION '(' string ')'           { CHECK_ERROR(http_dd_set_ssl_version(last_driver, $3), @3,
                                                             "curl: unsupported SSL version: %s", $3);
                                                 free($3); }
+    | KW_USE_SYSTEM_CERT_STORE '(' yesno ')'  {
+        const gchar *ca_dir = auto_detect_ca_dir();
+        const gchar *ca_file = auto_detect_ca_file();
+        if (!ca_dir && !ca_file) YYERROR;
+        http_dd_set_ca_file(last_driver, ca_file);
+        http_dd_set_ca_dir(last_driver, ca_dir);
+      }
```

Comment:
I think we should do this only when `use-system-cert-store()` is set to `yes`.

(In general, we should not use system certs by default, because - in my opinion - syslog-ng should not trust "everything on the web", our trust chain should be much narrower. Of course, there are exceptions, like `telegram()`, `slack()`, and #3078, but we can wrap those into an SCL block, e.g: `block destination telegram(use-system-cert-store(yes) ...)`)

Reply:
just wanted to open the pr before go home... so it's -of course- not intentional.

The default will be -of course- *no*.